tests/unit/Hooks/ImageHooksTest.php

```diff
@@ -0,0 +1,219 @@
+<?php
+/**
+ * ownCloud - mediametadata
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Jalpreet Singh Nanda (:imjalpreet) <jalpreetnanda@gmail.com>
+ * @copyright Jalpreet Singh Nanda (:imjalpreet) 2016
+ */
+
+namespace OCA\MediaMetadata\Hooks;
+
+function getimagesize($filename = null, array &$imageinfo = null) {
+	return ImageHooksTest::$dimensions ?: getimagesize($filename);
+}
+
+use OCA\MediaMetadata\AppInfo\Application;
+use OCA\MediaMetadata\Services\ImageDimension;
+use OCA\MediaMetadata\Services\ImageDimensionMapper;
+use OCA\MediaMetadata\Tests\TestCase;
+use OCP\ILogger;
+use OCP\IDb;
+use OCP\Files\File;
+use OCP\Files\Folder;
+use PHPUnit_Framework_TestCase;
+
+/**
+ * Class ImageHooksTest
+ *
+ * @package OCA\MediaMetadata\Hooks
+ */
+class ImageHooksTest extends PHPUnit_Framework_TestCase {
+	/**
+	 * @var \OCA\MediaMetadata\Hooks\ImageHooks
+	 */
+	protected $imagehooks;
+	/**
+	 * @var \PHPUnit_Framework_MockObject_MockObject|\OC\Files\Node\Root
+	 */
+	protected $root;
+	/**
+	 * @var \PHPUnit_Framework_MockObject_MockObject|\OCA\MediaMetadata\Services\ImageDimensionMapper
+	 */
+	protected $mapper;
+	/**
+	 * @var ILogger
+	 */
+	protected $logger;
+	/**
+	 * @var \PHPUnit_Framework_MockObject_MockObject|\OCP\IDb
+	 */
+	protected $db;
+	/**
+	 * @var \PHPUnit_Framework_MockObject_MockObject|\OCA\MediaMetadata\Services\ImageDimension
+	 */
+	protected $imageDimension;
+
+	/**
+	 * @var array $dimensions that will be returned by getimagesize()
+	 */
+	public static $dimensions;
+
+	protected function setUp() {
+		parent::setUp();
+
+		$app = new Application();
+		$container = $app->getContainer();
+
+		$this->root = $this->getMockBuilder('OC\Files\Node\Root')
+			->disableOriginalConstructor()
+			->getMock();
+		$this->db = $container->query('ServerContainer')->getDb();
+		$this->mapper = $this->mockImageDimensionMapper();
+		$this->logger = $this->getMockBuilder('\OCP\ILogger')
+			->disableOriginalConstructor()
+			->getMock();
+		$this->imageDimension = $this->mockImageDimension();
+		$this->imagehooks = new ImageHooks(
+			$this->root,
+			$this->mapper,
+			$container->query('ServerContainer')->getConfig()->getSystemValue('datadirectory')
+		);
+	}
+
+	public function testPostCreate() {
+		$location = 'testFolder/test.jpg';
+		$fileId = 260495;
+		$jpgFile = $this->mockJpgFile($location, $fileId);
+
+		$imageWidth = 100;
+		$imageHeight = 100;
+		$dimensions = array($imageWidth, $imageHeight);
+
+		$this->mockGetImageSize($this->imagehooks, $location, $dimensions);
+
+		$this->logger->log('debug', 'Test Image Path: '.$location, array('app' => 'MediaMetadata'));
+		$this->logger->log('debug', 'Test Image Height: '.$imageHeight, array('app' => 'MediaMetadata'));
+		$this->logger->log('debug', 'Test Image Width: '.$imageWidth, array('app' => 'MediaMetadata'));
+
+		$this->imageDimension->expects($this->once())
```

Comment:
There is a problem with the imageDimension mocked methods, because these don't exist, they're created on the fly. Would be worth checking in core to see how they mock `Entity`